vim/core/oni-plugin-git/index.js

```diff
@@ -1,48 +1,80 @@
-const fs = require("fs");
-const path = require("path");
-
-const activate = (Oni) => {
-
-    const gitBranchIndicator = Oni.statusBar.createItem(0, -3);
-
-    const pathIsDir = (p) => {
-        return new Promise((resolve, reject) => {
-            fs.stat(p, (error, stats) => {
-                if (error) {
-                    reject(error);
-                } else {
-                    resolve(stats.isDirectory());
-                }
-            });
-        })
+const fs = require('fs');
+const path = require('path');
+const { promisify } = require('util');
+const fsStat = promisify(fs.stat);
```

Comment:
Cool! I hadn't tried this yet, was actually looking for an example (will be helpful for the file explorer because of all the node-callback APIs)

Reply:
Think node added it a couple of months ago and I've been taking every opportunity I get to promisify things, that way and avoid all the boiler plate that comes with doing it manually.